### ISSUE TITLE
CompileTimeErrorCode.INSTANCE_ACCESS_TO_STATIC_MEMBER doesn't handle mixins well

### ISSUE THREAD

**Issue by stereotype441** [dart-lang/sdk#47452]
Consider the following code:
```dart
mixin A {
  static m() {}
}
f(A a) {
  a.m;
}
```
The analyzer currently produces the error message:
```
Static method 'm' can't be accessed through an instance. Try using the class 'A' to access the method.
```
It would be better to say `Try using the mixin 'A'...`.

There's a similar issue if the method in question is in an extension; I'm currently making a fix for that (https://dart-review.googlesource.com/c/sdk/+/216680), which is easy because extension elements can be distinguished by their ElementKind.  The corresponding fix for mixins will be more difficult because we don't have a separate ElementKind for mixins.

**Comment by bwilkerson** [comment]
You can use `ClassElement.isMixin` to identify mixins.

**Comment by stereotype441** [comment]
Fixed in 72daafd70c417dd5c790215e469affd21692ce29
